src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java

```diff
@@ -0,0 +1,162 @@
+            final Map<String, String> config) {
+        if (this.exists(topicName)) {
+            log.debug("Topic {} already exists, no need to create.", topicName);
+        } else {
+            this.createTopic(topicName, settings, config);
+        }
+    }
+
+    /**
+     * Describe the current configuration of a Kafka topic.
+     *
+     * @param topicName the topic name
+     * @return settings of topic including number of partitions and replicationFactor
+     */
+    public TopicSettings describe(final String topicName) {
+        try {
+            final Map<String, KafkaFuture<TopicDescription>> kafkaTopicMap =
+                    this.adminClient.describeTopics(List.of(topicName)).values();
+            final TopicDescription description =
+                    kafkaTopicMap.get(topicName).get(this.timeout.toSeconds(), TimeUnit.SECONDS);
+            final List<TopicPartitionInfo> partitions = description.partitions();
+            final int replicationFactor = partitions.stream()
+                    .findFirst()
+                    .map(TopicPartitionInfo::replicas)
+                    .map(List::size)
+                    .orElseThrow(() -> new IllegalStateException("Topic " + topicName + " has no partitions"));
+            return TopicSettings.builder()
+                    .replicationFactor((short) replicationFactor)
+                    .partitions(partitions.size())
+                    .build();
+        } catch (final InterruptedException | ExecutionException | TimeoutException e) {
+            throw new RuntimeException("Failed to retrieve description of topic " + topicName, e);
+        }
+    }
+
+    @Override
+    public void close() {
+        this.adminClient.close();
+    }
+
+    /**
+     * Check whether a Kafka topic exists.
+     *
+     * @param topicName the topic name
+     * @return whether a Kafka topic with the specified name exists or not
+     */
+    public boolean exists(final String topicName) {
+        try {
+            final Map<String, KafkaFuture<TopicDescription>> kafkaTopicMap =
+                    this.adminClient.describeTopics(List.of(topicName)).values();
+            kafkaTopicMap.get(topicName).get(this.timeout.toSeconds(), TimeUnit.SECONDS);
+            return true;
+        } catch (final ExecutionException e) {
+            if (e.getCause() instanceof UnknownTopicOrPartitionException) {
+                return false;
+            } else {
+                throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+            }
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+        } catch (final TimeoutException e) {
+            throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+        }
```

Comment:
couldn't you throw the exception just once after the try-catch block? All valid responses are in early returns anyway

Reply:
We have to handle the exceptions differently. InterruptedExceptions require Thread.currentThread().interrupt();